File: SimpleLinearRegression/SimpleLinearRegression.py
from scipy import stats
import pandas as pd
import matplotlib.pyplot as plt

# reads csv file and stores data in pandas data frame
weather = pd.read_csv("../data/weather.csv")

# gets columns to be used as x and y values for linear model
temperature = weather['temperature']
days = weather['days']

# fits a linear model to given data
slope, intercept, r_value, p_value, std_error = stats.linregress(x=days, y=temperature)

# equation of a line: y = bx + a
predict_y = slope * days + intercept

# scatter plot of given data
plt.scatter(x=days, y=temperature)

# set range of x-axis of plot
plt.xlim(0, 25)

# plot predicted line: 'k' is black and '-' is solid line
plt.plot(days, predict_y, 'k-')

# label plot title and axes
plt.title('Weather: Temperature vs Days')
plt.xlabel('days')
plt.ylabel('temperature')

# show plot
plt.show()

File: SimpleLinearRegression/SimpleLinearRegression.R
# imports library with graphing tools
library(ggplot2) 

# reads csv file and stores data in data frame
weather <- read.csv("data/weather.csv")

# fits a linear model to given data
model <- lm(temperature ~ days, data=weather)

# creates a scatterplot using given data
ggplot(aes(x=days, y=temperature), data=weather) +
	geom_point(shape=4) + # plots points and changes shape of points
	geom_smooth(method=lm, se=FALSE) + # plots linear best fit without confidence range
	ggtitle("Weather: Temperature vs Days")